src/kaltura-live-plugin.tsx

```diff
@@ -100,9 +107,42 @@ export class KalturaLivePlugin extends KalturaPlayer.core.BasePlugin implements
       this.eventManager.listen(this.player, this.player.Event.FIRST_PLAY, this._handleFirstPlay);
       this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA, this.handleTimedMetadata);
       this.eventManager.listen(this.player, this.player.Event.MEDIA_LOADED, this._handleMediaLoaded);
+      this.eventManager.listen(this.player, this.player.Event.PLAYER_STATE_CHANGED, this._handlePlayerStateChange);
+    }
+  };
+
+  private _resetBufferingTimeout = () => {
+    this.logger.debug('Reset buffering timeout');
+    if (this._bufferingTimeout) {
+      clearTimeout(this._bufferingTimeout);
+      this._bufferingTimeout = null;
+    }
+  };
+
+  private _handlePlayerStateChange = ({payload}: {payload: {newState: {type: typeof StateType}; oldState: {type: typeof StateType}}}) => {
+    if (this._bufferingTimeout && payload.newState.type === StateType.PLAYING && payload.oldState.type === StateType.BUFFERING) {
```

Comment:
What if we are out of buffering and new state is not playing - we should try to delete the timeout anyway. For example, we are in buffering and user clicks - pause.